### PR TITLE
JSON exercise

### PR DESCRIPTION
**Pull request by consta35**

### CODE REVIEW

File: json_exerciseAnswers.ipynb [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+      "  if __name__ == '__main__':\n"
+     ]
+    }
+   ],
+   "source": [
+    "#sort the data based on code number\n",
+    "dfs = df.sort('code')\n",
+    "\n",
+    "#replace the empty spaces with NaN\n",
+    "dfs['name'].replace('', np.nan, inplace=True)\n",
```
Comment (amirziai) [inline json_exerciseAnswers.ipynb]: Using `inplace` is not a good idea, always try to [avoid it](http://stackoverflow.com/questions/22532302/pandas-peculiar-performance-drop-for-inplace-rename-after-dropna).

File: json_exerciseAnswers.ipynb [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+    "\n",
+    "#use .fillna forward fill to fill in the theme names\n",
+    "fill = dfs.fillna(method = 'ffill')\n",
```
Comment (amirziai) [inline json_exerciseAnswers.ipynb]: Is [this](https://gist.github.com/amirziai/f29686ab7d8d2da70d1020e34609f0f3) helpful?

File: json_exerciseAnswers.ipynb [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Question 3:\n",
+    "#read in the data as a json\n",
+    "data = json.load((open('data/world_bank_projects.json')))\n",
```
Comment (amirziai) [inline json_exerciseAnswers.ipynb:548]: A general point:
[DRY](https://www.wikiwand.com/en/Don't_repeat_yourself) is a very useful concept in software development. Every time you write something more than once you should think about it. It's easy to overdo it too so don't worry about it for this exercise but just be aware of its existence.

**Review by amirziai — CHANGES_REQUESTED** [review]
Looks pretty good 👍 

Added a couple of comments.

**Comment by amir-ziai-zefr** [comment]
You're joining the dataframe with itself, which is not giving you any new information.

As you suggested you need to create a lookup table that maps codes to names. One way to do it is:
`df[df.name.str.len() > 0].groupby(['code']).first().reset_index()`.

Does that make sense? Can you now get the final answer? Is this efficient? What could go wrong?

**Comment by amir-ziai-zefr** [comment]
Also take a look at [drop_duplicates](http://pandas.pydata.org/pandas-docs/version/0.17.1/generated/pandas.DataFrame.drop_duplicates.html)

**Review by amirziai — APPROVED** [review]

**Pull request merged**